jobs/FunctionTest/FunctionTest.groovy

```diff
@@ -103,23 +133,17 @@ def functionTest(String test_name, String label_name, String TEST_GROUP, Boolean
                                 popd
                                 '''
 
-                                // Get main test scripts for un-manifest-src test
-                                if (this.TEST_TYPE != "manifest") {
-                                    def exists = fileExists 'RackHD'
-                                    if( !exists ){
-                                        echo "Checkout RackHD for un-src test."
-                                        def url = "https://github.com/RackHD/RackHD.git"
-                                        def branch = "master"
-                                        def targetDir = "RackHD"
-                                        shareMethod.checkout(url, branch, targetDir)
-                                    }
-                                }
                                 // Run smoke test
                                 sh './build-config/test.sh'
                             }
                         } catch(error){
                             throw error
                         } finally{
+                            // Clean up test stack
```

Comment:
"$artifact_dir/*.*" may be null if test failed, and stash null file will throw error. Then cleanup won't run.